UI/src/app/home/home.component.ts

```diff
@@ -58,13 +58,23 @@ export class HomeComponent implements AfterViewInit, OnInit {
             console.log("Plot was pressed");
             console.log(this.textInput.getTextInput());
             console.log(this.dataset.name);
-            this.backendConnector.requestData(this.textInput.getTextInput(), this.dataset.id);
-            setTimeout( () => {
+            console.log(this.chart);
+            this.chart.plot();
+            let promise = this.backendConnector.requestData(this.textInput.getTextInput(), this.dataset.name, this);
+            promise.then(function(err){
+                console.log(err);
+                throw err;
+            }, function(that){
+                console.log(that.chart);
+                that.chart.plot();
```

Comment:
shouldnt they be the other way around?

the first attribute of `then`is called `onfulfilled` and the secound is called `onrejected`